adapter/internal/oasparser/envoyconf/routes_with_clusters.go

```diff
@@ -126,15 +129,24 @@ func CreateRoutesWithClusters(mgwSwagger model.MgwSwagger, upstreamCerts map[str
 	}
 	// check API level sandbox endpoints available
 	if mgwSwagger.GetSandEndpoints() != nil && len(mgwSwagger.GetSandEndpoints().Endpoints) > 0 {
+		selectedBasePathSand := apiLevelBasePathProd
 		apiLevelEndpointSand := mgwSwagger.GetSandEndpoints()
-		if apiLevelbasePath == "" {
-			apiLevelbasePath = strings.TrimSuffix(apiLevelEndpointSand.Endpoints[0].Basepath, "/")
+		if apiLevelBasePathProd == "" {
+			// no production endpoint, assign sandbox endpoint basepath as apiLevelbasePath
+			apiLevelBasePathProd = strings.TrimSuffix(apiLevelEndpointSand.Endpoints[0].Basepath, "/")
+			selectedBasePathSand = apiLevelBasePathProd
```

Comment:
What if there is a prod endpoint that does not contain a basepath?
for example:
- prod: http://prod.com
- sand: http://sand.com/api/v1

is it set `/api/v1` as basepath for prod?

Reply:
It is set to `/api/v1` with the current implementation of this PR. I'll update the if condition to check for both `apiLevelBasePathProd` and `apiLevelClusterNameProd`. (`apiLevelBasePathProd && apiLevelClusterNameProd`)

Reply:
fixed